File: julia8024/PGS_42883.py
def solution(number, k):
    stack = []
    
    for val in number:
        if len(stack) == 0:
            stack.append(val)
            continue
        if k > 0:
            while stack[-1] < val:
                stack.pop()
                k -= 1
                if len(stack) == 0 or k <= 0:
                    break
        stack.append(val)
        
    return "".join(stack[:-k] if k > 0 else stack)

File: chaeryeon823/PGS_42883.js
// 큰 수 만들기
// lv 2
// 그리디
// 스택

function solution(number, k) {
  const stack = [];

  for (let digit of number) {
    while (k > 0 && stack.length && stack[stack.length - 1] < digit) {
      stack.pop();
      k--;
    }
    stack.push(digit);
  }

  return stack.slice(0, stack.length - k).join("");
}
